crates/bifrost/src/loglet/mod.rs

```diff
@@ -156,21 +156,40 @@ pub type SendableLogletReadStream = Pin<Box<dyn LogletReadStream + Send>>;
 
 #[allow(dead_code)]
 pub(crate) struct LogletCommitResolver {
-    tx: oneshot::Sender<Result<LogletOffset, AppendError>>,
+    tx: Option<oneshot::Sender<Result<LogletOffset, AppendError>>>,
 }
 
 #[allow(dead_code)]
 impl LogletCommitResolver {
-    pub fn sealed(self) {
-        let _ = self.tx.send(Err(AppendError::Sealed));
+    pub fn sealed(mut self) {
+        let _ = self
+            .tx
+            .take()
+            .expect("must be set")
+            .send(Err(AppendError::Sealed));
     }
 
-    pub fn offset(self, offset: LogletOffset) {
-        let _ = self.tx.send(Ok(offset));
+    pub fn offset(mut self, offset: LogletOffset) {
+        let _ = self.tx.take().expect("must be set").send(Ok(offset));
     }
 
-    pub fn error(self, err: AppendError) {
-        let _ = self.tx.send(Err(err));
+    pub fn error(mut self, err: AppendError) {
+        let _ = self.tx.take().expect("must be set").send(Err(err));
+    }
+}
+
+#[derive(Debug, Clone, Copy, thiserror::Error)]
+#[error("Commit resolver was dropped")]
+struct CommitCancelled;
+
+/// If a LogletCommitResolver is dropped without being
+/// 'resolved', we resolve it automatically as being cancelled
+/// To make it distinguished from a Shutdown.
+impl Drop for LogletCommitResolver {
```

Comment:
I am not sure this is necessary. Since if the LogletCommitResolver is dropped and the sender channel is closed, this causes a RecvErr to be received at the receiver side which is then mapped into an AppendError::Shutdown.

I am wondering if a Shutdown is what we need to return here. This still can happen for example if a connection is lost in case of remote sequencer.

What do you think @AhmedSoliman

Reply:
Perhaps we can map the RecvErr to a more descriptive type. For instance, `CommitStatusUnknown` and mark it as retryable append error?